### ISSUE TITLE
Encoding error if size != 256 in Unity

### ISSUE THREAD

**Issue by mtschoen-unity** [micjahn/ZXing.Net#183]
I tried to use a 512x512 texture, and this was the result. The same code works as expected if the size is 256x256.

![image](https://user-images.githubusercontent.com/19391263/56764477-9b7e3280-6759-11e9-98dd-66e3a8cc88a8.png)

**Comment by micjahn** [comment]
I can't reproduce this behavior.
Do you have a sample project which you can provide? Or a code snippet which shows the generation of the bitmap?
What version of ZXing.Net do you use?

**Comment by o-kniejski** [comment]
I have the same symptom. All Base2 sizes starting from 512 produce similar results, sizes in between "smear" the picture.

I tested the zxing.unity.dll in Clients/UnityDemo/Assets/ as well as the most recent version in release ZXing.Net 0.16.5
I am using Unity 2019.2

Sample Code:
`public class QRWrite : MonoBehaviour
{
    public Texture2D qrCode;
    public int dimensions = 256;
    public string qrText = "FooBar";

    void OnGUI()
    {
        // text inputfield
        qrText = GUI.TextField(new Rect(10, 10, 200, 20), qrText);

        // button
        if (GUI.Button(new Rect(210, 10, 50, 20), "Encode"))
        {
            qrCode = generateQR(qrText, dimensions, dimensions);
        }

        // qr image
        if (GUI.Button(new Rect(10, 50, dimensions,dimensions), qrCode)) { }
    }

    private static Texture2D generateQR(string text, int width, int height)
    {
        Texture2D encoded = new Texture2D(width, height);
        var writer = new BarcodeWriter
        {
            Format = BarcodeFormat.QR_CODE,
            Options = new QrCodeEncodingOptions
            {
                Height = height,
                Width = width
            }
        };
        var color32 = writer.Write(text);

        encoded.SetPixels32(color32);
        encoded.Apply();

        return encoded;
    }

}`

**Comment by micjahn** [comment]
Can you please try the following:
```
using ZXing;
using ZXing.Rendering;

public class QRWrite : MonoBehaviour
{
    public Texture2D qrCode;
    public int dimensions = 256;
    public string qrText = "FooBar";

    void OnGUI()
    {
        // text inputfield
        qrText = GUI.TextField(new Rect(10, 10, 200, 20), qrText);

        // button
        if (GUI.Button(new Rect(210, 10, 50, 20), "Encode"))
        {
            qrCode = generateQR(qrText, dimensions, dimensions);
        }

        // qr image
        if (GUI.Button(new Rect(10, 50, dimensions,dimensions), qrCode)) { }
    }

    private static Texture2D generateQR(string text, int width, int height)
    {
        Texture2D encoded = new Texture2D(width, height);
        var writer = new BarcodeWriterPixelData
        {
            Format = BarcodeFormat.QR_CODE,
            Options = new QrCodeEncodingOptions
            {
                Height = height,
                Width = width
            }
        };
        var pixelData = writer.Write(text);

        encoded.SetPixels32(pixelData.ToColor32());
        encoded.Apply();

        return encoded;
    }
}
```

**Comment by o-kniejski** [comment]
Thanks for the quick reply!
Sadly, ToColor32() is not ok
```
encoded.SetPixels32(pixelData.ToColor32());
```
"PixelData" enthält keine Definition für "ToColor32", und es konnte keine zugängliche ToColor32 Erweiterungsmethode gefunden werden [...]

There is no ToColor32 for PixelData.
My IDE marks using ZXing.Rendering as unused...

**Comment by micjahn** [comment]
Are you sure? I looked inside zxing.unity.dll and the method exists:
![grafik](https://user-images.githubusercontent.com/3475026/66079793-0d5a7380-e565-11e9-9d14-c40b626f6a01.png)

**Comment by o-kniejski** [comment]
I found the problem...
You are correct, zxing.unity.dll 0.16.5.0 contains this method.
Version 0.15.0.0, which is included in ZXing.Net.DemoClients.0.16.5.0.zip does not :)
Thanks for the swift help!

**Comment by micjahn** [comment]
Bug was fixed with version 0.16.4
